### PR TITLE
LPS-132895 | master

### PR DESCRIPTION
**Pull request by Tim-Cao**
https://issues.liferay.com/browse/LPS-132895

This is only for master.

### CODE REVIEW

**Comment by Tim-Cao** [comment]
ci:test:sf

**Comment by Tim-Cao** [comment]
ci:test:echo:nocompile

**Comment by Tim-Cao** [comment]
See https://github.com/liferay-echo/liferay-portal/pull/4697

**Comment by Tim-Cao** [comment]
ci:close

File: portal-web/test/functional/com/liferay/portalweb/macros/layout/pageeditor/PageEditor.macro [hunk 1]
```diff
@@ -73,76 +73,146 @@ definition {
...
+					key_position = "${position}",
+					locator1 = "PageEditor#FRAGMENTS_AND_WIDGETS_TAB_FRAGMENT",
+					locator2 = "PageEditor#GRID_COLUMN",
+					value1 = "");
+			}
 		}
-		else if ("${navTab}" == "Widgets") {
-			Navigator.gotoNavTab(navTab = "${navTab}");
 
+		if (("${navTab}" == "Widgets") && ("${displayStyle}" == "card")) {
```
Comment (david-gutierrez-mesa) [inline portal-web/test/functional/com/liferay/portalweb/macros/layout/pageeditor/PageEditor.macro:162]: Shouldn't we add here ("${navTab}" == "Fragments") to keep the else?

File: portal-web/test/functional/com/liferay/portalweb/macros/layout/pageeditor/PageEditor.macro [hunk 2]
```diff
@@ -73,76 +73,146 @@ definition {
...
-				}
-				else if ("${height}" == "Tall") {
-					DragAndDrop.javaScriptDragAndDropObjectBelowTallNestedObject(
-						key_portletName = "${portletName}",
-						key_targetFragmentName = "${targetFragmentName}",
-						locator1 = "PageEditor#FRAGMENTS_AND_WIDGETS_TAB_WIDGET",
-						locator2 = "PageEditor#ANY_NESTED_FRAGMENT_DROP_TARGET",
-						value1 = "${portletName}");
-				}
+			if ((isSet(targetFragmentName)) && ("${height}" == "Short")) {
```
Comment (david-gutierrez-mesa) [inline portal-web/test/functional/com/liferay/portalweb/macros/layout/pageeditor/PageEditor.macro:167]: Can we remove this nested if adding conditions?

File: portal-web/test/functional/com/liferay/portalweb/macros/layout/pageeditor/PageEditor.macro [hunk 3]
```diff
@@ -73,76 +73,146 @@ definition {
...
+					key_portletName = "${portletName}",
+					key_position = "${position}",
+					locator1 = "PageEditor#FRAGMENTS_AND_WIDGETS_TAB_WIDGET_CARD_ITEM",
+					locator2 = "PageEditor#GRID_COLUMN",
+					value1 = "${portletName}");
+			}
+		}
+
+		if (("${navTab}" == "Widgets") && ("${displayStyle}" == "list")) {
+			if ((isSet(targetFragmentName)) && ("${height}" == "Short")) {
```
Comment (david-gutierrez-mesa) [inline portal-web/test/functional/com/liferay/portalweb/macros/layout/pageeditor/PageEditor.macro:197]: Can we remove this nested if adding conditions?

File: portal-web/test/functional/com/liferay/portalweb/macros/layout/pageeditor/PageEditor.macro [hunk 4]
```diff
@@ -445,20 +641,84 @@ definition {
...
+				DragAndDrop.javaScriptDragAndDropToBottom(
+					key_collectionName = "${collectionName}",
+					key_fragmentName = "${fragmentName}",
+					key_targetFragmentName = "${targetFragmentName}",
+					locator1 = "PageEditor#FRAGMENTS_AND_WIDGETS_TAB_FRAGMENT_CARD_ITEM",
+					locator2 = "PageEditor#ANY_FRAGMENT_DROP_TARGET",
+					value1 = "");
+			}
+
+			if ((!(isSet(scope))) && (!(isSet(targetFragmentName)))) {
```
Comment (david-gutierrez-mesa) [inline portal-web/test/functional/com/liferay/portalweb/macros/layout/pageeditor/PageEditor.macro:701]: Can we remove this nested if adding conditions?

File: portal-web/test/functional/com/liferay/portalweb/macros/layout/pageeditor/PageEditor.macro [hunk 5]
```diff
@@ -445,20 +641,84 @@ definition {
...
+					key_collectionName = "${collectionName}",
+					key_fragmentName = "${fragmentName}",
+					locator1 = "PageEditor#FRAGMENTS_AND_WIDGETS_TAB_FRAGMENT_CARD_ITEM",
+					locator2 = "PageEditor#FRAGMENT_DROP_TARGET",
+					value1 = "");
+			}
+		}
+
+		if ("${displayStyle}" == "list") {
+			if (("${scope}" == "Master") && ("${dropTarget}" == "Header")) {
```
Comment (david-gutierrez-mesa) [inline portal-web/test/functional/com/liferay/portalweb/macros/layout/pageeditor/PageEditor.macro:712]: Can we remove this nested if adding conditions?

File: portal-web/test/functional/com/liferay/portalweb/macros/layout/pageeditor/PageEditor.macro [hunk 6]
```diff
@@ -445,20 +641,84 @@ definition {
...
 				DragAndDrop.javaScriptDragAndDropToUp(
 					key_collectionName = "${collectionName}",
 					key_fragmentName = "${fragmentName}",
 					locator1 = "PageEditor#FRAGMENTS_AND_WIDGETS_TAB_FRAGMENT",
 					locator2 = "PageEditor#HEADER_DROP_TARGET",
 					value1 = "");
 			}
-			else if ("${dropTarget}" == "Footer") {
+
+			if (("${scope}" == "Master") && ("${dropTarget}" == "Footer")) {
```
Comment (david-gutierrez-mesa) [inline portal-web/test/functional/com/liferay/portalweb/macros/layout/pageeditor/PageEditor.macro:721]: Can we remove this nested if adding conditions?

File: portal-web/test/functional/com/liferay/portalweb/macros/layout/pageeditor/PageEditor.macro [hunk 7]
```diff
@@ -467,17 +727,17 @@ definition {
 					locator2 = "PageEditor#ANY_FRAGMENT_DROP_TARGET",
 					value1 = "");
 			}
-		}
-		else if (isSet(targetFragmentName)) {
-			if ("${dropTarget}" == "Header") {
+
+			if ((isSet(targetFragmentName)) && ("${dropTarget}" == "Header")) {
```
Comment (david-gutierrez-mesa) [inline portal-web/test/functional/com/liferay/portalweb/macros/layout/pageeditor/PageEditor.macro:731]: Can we remove this nested if adding conditions?

File: portal-web/test/functional/com/liferay/portalweb/macros/layout/pageeditor/PageEditor.macro [hunk 8]
```diff
@@ -486,73 +746,73 @@ definition {
...
-		}
-		else {
-			DragAndDrop.javaScriptDragAndDropToObject(
-				key_collectionName = "${collectionName}",
-				key_fragmentName = "${fragmentName}",
-				locator1 = "PageEditor#FRAGMENTS_AND_WIDGETS_TAB_FRAGMENT",
-				locator2 = "PageEditor#FRAGMENT_DROP_TARGET",
-				value1 = "");
+
+			if ((!(isSet(scope))) && (!(isSet(targetFragmentName)))) {
```
Comment (david-gutierrez-mesa) [inline portal-web/test/functional/com/liferay/portalweb/macros/layout/pageeditor/PageEditor.macro:750]: Can we remove this nested if adding conditions?

File: portal-web/test/functional/com/liferay/portalweb/macros/layout/pageeditor/PageEditor.macro [hunk 9]
```diff
@@ -486,73 +746,73 @@ definition {
...
 
 		if ("${collectionName}" == "Layout Elements") {
-			if ("${fragmentName}" == "Grid") {
-				if (isSet(position)) {
-					var key_position = "${position}";
-				}
-				else {
-					var key_position = "1";
-				}
+			if (!(isSet(position))) {
```
Comment (david-gutierrez-mesa) [inline portal-web/test/functional/com/liferay/portalweb/macros/layout/pageeditor/PageEditor.macro:763]: Can we remove this nested if adding conditions?

File: portal-web/test/functional/com/liferay/portalweb/macros/layout/pageeditor/PageEditor.macro [hunk 10]
```diff
@@ -486,73 +746,73 @@ definition {
...
+				var key_position = "1";
+			}
 
+			if ("${fragmentName}" == "Grid") {
```
Comment (david-gutierrez-mesa) [inline portal-web/test/functional/com/liferay/portalweb/macros/layout/pageeditor/PageEditor.macro:767]: Can we remove this nested if adding conditions?

File: portal-web/test/functional/com/liferay/portalweb/macros/layout/pageeditor/PageEditor.macro [hunk 11]
```diff
@@ -486,73 +746,73 @@ definition {
...
+		if ("${collectionName}" == "Content Display") {
 			if ("${fragmentName}" == "Collection Display") {
 				AssertElementPresent(
 					key_fragmentName = "${fragmentName}",
 					key_message = "No Collection Selected Yet",
 					locator1 = "Fragment#NOT_COLLECTION_SELECTED_MESSAGE");
 			}
-			else if ("${fragmentName}" == "Content Display") {
+
+			if ("${fragmentName}" == "Content Display") {
```
Comment (david-gutierrez-mesa) [inline portal-web/test/functional/com/liferay/portalweb/macros/layout/pageeditor/PageEditor.macro:790]: Can we remove this nested if adding conditions?

File: portal-web/test/functional/com/liferay/portalweb/macros/layout/pageeditor/PageEditor.macro [hunk 12]
```diff
@@ -486,73 +746,73 @@ definition {
...
+
+			if (("${fragmentName}" != "Collection Display") && ("${fragmentName}" != "Content Display") && ("${fragmentName}" != "Display Page Content")) {
 				AssertElementPresent(
 					key_fragmentName = "${fragmentName}",
 					locator1 = "Fragment#FRAGMENT_HEADER");
 			}
 		}
-		else if ("${composition}" == "true") {
+
+		if ("${composition}" == "true") {
```
Comment (david-gutierrez-mesa) [inline portal-web/test/functional/com/liferay/portalweb/macros/layout/pageeditor/PageEditor.macro:811]: Are we missing here the else with a negated condition?

**Review by david-gutierrez-mesa — COMMENTED** [review]
Review done

**Comment by Tim-Cao** [comment]
Thanks for your reviewing @david-gutierrez-mesa, I made some changes see
https://github.com/liferay-echo/liferay-portal/pull/4697/commits/ebee440136818b1fe7e9d8fd380f079af5c5c2cd
and 
https://github.com/liferay-echo/liferay-portal/pull/4697/commits/3a68aa837e36b8ab9991323390e0363b44c229e0
Btw, I don't think poshi support switch loop.

**Pull request closed without merging**